tests/samples_test.py

```diff
@@ -0,0 +1,420 @@
+    "NestedClenshawCurtis [3]",
+    "SlowNestedClenshawCurtis [0]",
+    "SlowNestedClenshawCurtis [1]",
+    "SlowNestedClenshawCurtis [2]",
+    "SlowNestedClenshawCurtis [3]",
+    "SlowNestedClenshawCurtis [4]",
+    "SlowNestedClenshawCurtis [5]",
+    "NestedTrigonometric [0]",
+    "NestedTrigonometric [1]",
+    "NestedTrigonometric [2]",
+]
+
+
+def test_clenshaw_initial():
+    """test default properties"""
+    f = smolyay.samples.ClenshawCurtisPointSet([-2, 1], 7)
+    assert numpy.array_equal(f.domain, [-2, 1])
+    assert f.degree == 7
+
+
+def test_trig_initial():
+    """test default properties"""
+    f = smolyay.samples.TrigonometricPointSet([0, 4 * numpy.pi], 7)
+    assert numpy.array_equal(f.domain, [0, 4 * numpy.pi])
+    assert f.frequency == 7
+
+
+@pytest.mark.parametrize(
+    "nested_samples",
+    [
+        smolyay.samples.NestedClenshawCurtisPointSet,
+        smolyay.samples.SlowNestedClenshawCurtisPointSet,
+        smolyay.samples.NestedTrigonometricPointSet,
+    ],
+    ids=["NestedClenshawCurtis", "SlowNestedClenshawCurtis", "NestedTrigonometric"],
+)
+def test_nested_initial(nested_samples):
+    """test default properties"""
+    f = nested_samples([-10, 10], 7)
+    assert numpy.array_equal(f.domain, [-10, 10])
+    assert f.max_level == 7
+
+
+@pytest.mark.parametrize(
+    "samples",
+    [
+        smolyay.samples.ClenshawCurtisPointSet,
+        smolyay.samples.TrigonometricPointSet,
+        smolyay.samples.NestedClenshawCurtisPointSet,
+        smolyay.samples.SlowNestedClenshawCurtisPointSet,
+        smolyay.samples.NestedTrigonometricPointSet,
+    ],
+    ids=[
+        "ClenshawCurtis",
+        "Trigonometric",
+        "NestedClenshawCurtis",
+        "SlowNestedClenshawCurtis",
+        "NestedTrigonometric",
+    ],
+)
+def test_domain_error(samples):
+    """test default properties"""
+    with pytest.raises(TypeError):
+        samples([-10, 10, 20], 7)
+    with pytest.raises(TypeError):
+        samples([[-10, 10], [-10, 10]], 7)
+
+
+@pytest.mark.parametrize("samples,points", sample_points_answers, ids=sample_points_ids)
+def test_generate_points(samples, points):
+    """test the points of initialized UnidimensionalPointSet"""
+    assert numpy.allclose(samples.points, points, atol=1e-10)
+
+
+@pytest.mark.parametrize("samples,points", sample_points_answers, ids=sample_points_ids)
+def test_generate_points_set_domain(samples, points):
+    """test point correctness after setting a new domain"""
+    new_points = numpy.array(points) * 5 + 10
+    new_domain = samples.domain * 5 + 10
+    samples.domain = new_domain
+    assert numpy.array_equal(samples.domain, new_domain)
+    assert numpy.allclose(samples.points, new_points, atol=1e-10)
+
+
+@pytest.mark.parametrize("samples,points", sample_points_answers, ids=sample_points_ids)
+def test_len(samples, points):
+    """test the len method of UnidimensionalPointSet objects"""
+    assert len(samples) == len(points)
```

Comment:
These are all related, so I would combine these three tests to:
1. Check the length of samples match.
2. Check the points match before the domain is rescaled.
3. Check the points match after the domain is rescaled.